python/ray/experimental/gpu_object_manager/gpu_object_store.py

```diff
@@ -63,21 +67,33 @@ def __ray_recv__(
     communicator_name: str,
     obj_id: str,
     src_rank: int,
-    tensor_meta: List[Tuple["torch.Size", "torch.dtype"]],
+    tensor_meta: TensorMetadata,
 ):
     """Helper function that runs on the dst actor to receive tensors from the src actor."""
     from ray._private.worker import global_worker
 
     backend = collective.get_group_handle(communicator_name).backend()
+
     device = COLLECTIVE_BACKEND_TO_TORCH_DEVICE[backend]
+    tensor_basic_meta = tensor_meta.tensor_basic_meta
+    nixl_serialized_descs = tensor_meta.nixl_serialized_descs
+    nixl_agent_meta = tensor_meta.nixl_agent_meta
 
     gpu_object_store = global_worker.gpu_object_manager.gpu_object_store
     tensors = []
-    for meta in tensor_meta:
+    for meta in tensor_basic_meta:
         shape, dtype = meta
         tensor = torch.zeros(shape, dtype=dtype, device=device)
-        collective.recv(tensor, src_rank, group_name=communicator_name)
         tensors.append(tensor)
+
+    metadata = TensorTransportMetadata(
```

Comment:
How about removing `ray.util.collective.types.TensorTransportMetadata` and fusing it and `ray.experimental.gpu_object_manager.gpu_object_manager.TensorMetadata` into one?

```python
@dataclass
class TensorTransportMetadata
    src_rank: int
    tensor_basic_meta: List[Tuple["torch.Size", "torch.dtype"]]
    nixl_serialized_descs: Optional[bytes] = None
    nixl_agent_meta: Optional[bytes] = None

```